### DISCUSSION TITLE
What would be a good way to render emails, or embed the preview server inside another app?

### DISCUSSION THREAD

**Discussion by bhuynhdev** [shellscape/jsx-email#205] [Q&A]
I am building an admin dashboard with React/Remix and want to have the capability to preview the emails we've written with jsx-email right inside the app, and maybe potentially a small preview of the email before sending out.

What would be the best way to embed the preview server, or render the JSX to HTML (which can then be `__dangerouslySetInnerHTML`) so that we can have all functionalities in our big dashboard application? The preview server is awesome, but we're hoping we could avoid running another app in another port and switching browser tab.

I saw that React.email has a `render` utility, but I am really sold on jsx-email's mission, so what would everyone suggest us?

**Comment by shellscape** [comment]
Thanks for the support!

Is this going to be an app where the user can change the template and see the results in realtime? Or will it simply display templates (which are for all intents static in this scenario)?

**Reply by bhuynhdev** [reply]
I can work with the latter: just displaying static rendering result of templates
